File: framework/languages/dotnet/e2e/RegistrationCodec/Shared/Messages.cs
using MessagePack;
using Zlink.Framework.Contracts.Handlers;

namespace RegistrationCodec.Shared;

public static class RegistrationCodecNames
{
    public const string Channel = "reg-codec";
}

[ZLinkPacket("EchoAttrReq")]
public sealed record EchoAttrReq(string Value);

public sealed record EchoRes(string Value, string ContentType);

[ZLinkPacket("EchoAttrMsg")]
public sealed record EchoAttrMsg(string CommandId, string Value);

[ZLinkPacket("EchoDiReq")]
public sealed record EchoDiReq(string Value);

[ZLinkPacket("EchoAutoReq")]
public sealed record EchoAutoReq(string Value);

[ZLinkPacket("EchoAutoMsg")]
public sealed record EchoAutoMsg(string CommandId, string Value);

[ZLinkPacket("EchoManualReq")]
public sealed record EchoManualReq(string Value);

[ZLinkPacket("EchoManualMsg")]
public sealed record EchoManualMsg(string CommandId, string Value);

[ZLinkPacket("JsonEchoReq")]
public sealed record JsonEchoReq(string Value);

[ZLinkPacket("JsonEchoMsg")]
public sealed record JsonEchoMsg(string CommandId, string Value);

public sealed record CodecMismatchProbeRes(
    bool Rejected,
    string? FailureType,
    string? Value);

[ZLinkPacket("JsonGoldenReq")]
public sealed record JsonGoldenReq(
    string DisplayName,
    string Status,
    long Balance,
    byte[] Payload,
    int Score,
    double Ratio,
    string? OptionalNote);

public sealed record JsonGoldenRes(
    string DisplayName,
    string Status,
    long Balance,
    byte[] Payload,
    int Score,
    double Ratio,
    string? OptionalNote,
    string ContentType);

public sealed record EvidenceWaitReq(
    string[] ContainsAll,
    int TimeoutMilliseconds = 10000);

[MessagePackObject]
[ZLinkPacket("PackedEchoReq")]
public sealed class PackedEchoReq
{
    [Key(0)] public string Value { get; set; } = string.Empty;
}

[MessagePackObject]
[ZLinkPacket("PackedEchoRes")]
public sealed class PackedEchoRes
{
    [Key(0)] public string Value { get; set; } = string.Empty;
}

[MessagePackObject]
[ZLinkPacket("PackedEchoMsg")]
public sealed class PackedEchoMsg
{
    [Key(0)] public string CommandId { get; set; } = string.Empty;

    [Key(1)] public string Value { get; set; } = string.Empty;
}

File: framework/languages/node/e2e/RegistrationCodec/Shared/messages.ts
import { ZLinkPacket } from '@zlink-systems/framework';

export const RegistrationCodecNames = {
  channel: 'reg-codec'
} as const;

export const PacketNames = {
  echoAutoReq: 'EchoAutoReq',
  echoAutoMsg: 'EchoAutoMsg',
  echoAttrReq: 'EchoAttrReq',
  echoAttrMsg: 'EchoAttrMsg',
  echoManualReq: 'EchoManualReq',
  echoManualMsg: 'EchoManualMsg',
  echoDiReq: 'EchoDiReq',
  echoJsonReq: 'EchoJsonReq',
  echoJsonMsg: 'EchoJsonMsg',
  echoProtobufReq: 'EchoProtobufReq',
  echoProtobufMsg: 'EchoProtobufMsg',
  echoMessagePackReq: 'EchoMessagePackReq',
  echoMessagePackMsg: 'EchoMessagePackMsg'
} as const;

export interface EchoReq {
  readonly value: string;
}

export interface EchoRes {
  readonly value: string;
  readonly contentType: string;
}

export interface EchoMsg {
  readonly commandId: string;
  readonly value: string;
}

export class EchoAutoReq { constructor(readonly value: string) {} }
export class EchoAutoMsg { constructor(readonly commandId: string, readonly value: string) {} }
export class EchoAttrReq { constructor(readonly value: string) {} }
export class EchoAttrMsg { constructor(readonly commandId: string, readonly value: string) {} }
export class EchoManualReq { constructor(readonly value: string) {} }
export class EchoManualMsg { constructor(readonly commandId: string, readonly value: string) {} }
export class EchoDiReq { constructor(readonly value: string) {} }
export class EchoJsonReq { constructor(readonly value: string) {} }
export class EchoJsonMsg { constructor(readonly commandId: string, readonly value: string) {} }

@ZLinkPacket(PacketNames.echoProtobufReq)
export class ProtobufEchoReq {
  constructor(readonly value: string) {}
}

export class ProtobufEchoRes {
  constructor(readonly value: string) {}
}

@ZLinkPacket(PacketNames.echoProtobufMsg)
export class ProtobufEchoMsg {
  constructor(readonly value: string) {}
}

@ZLinkPacket(PacketNames.echoMessagePackReq)
export class MessagePackEchoReq {
  constructor(readonly value: string) {}
}

export class MessagePackEchoRes {
  constructor(readonly value: string) {}
}

@ZLinkPacket(PacketNames.echoMessagePackMsg)
export class MessagePackEchoMsg {
  constructor(
    readonly commandId: string,
    readonly value: string
  ) {}
}

export interface CodecScenarioRes {
  readonly value?: string;
  readonly contentType?: string;
  readonly json?: EchoRes;
  readonly protobufValue?: string;
  readonly messagePackValue?: string;
}

export interface EvidenceWaitReq {
  readonly containsAll: readonly string[];
  readonly timeoutMilliseconds?: number;
}
